pkg/config/model/warnings.go

```diff
@@ -10,3 +10,16 @@ type Warnings struct {
 	TraceMallocEnabledWithPy2 bool
 	Errors                    []error
 }
+
+// NewWarnings creates a new Warnings instance
+func (w *Warnings) NewWarnings(errors []error) *Warnings {
+	return &Warnings{
+		TraceMallocEnabledWithPy2: w.TraceMallocEnabledWithPy2,
+		Errors:                    append(w.Errors, errors...),
+	}
+}
```

Comment:
It would be more idiomatic to make this constructor a stand-alone function instead of a member function:

```suggestion
func NewWarnings(errors []error) *Warnings {
	return &Warnings{
		TraceMallocEnabledWithPy2: false,
		Errors:                    errors,
	}
}
```